core/xml.js

```diff
@@ -551,6 +576,21 @@ Blockly.Xml.domToBlockHeadless_ = function(xmlBlock, workspace) {
         // Fall through.
       case 'field':
         var field = block.getField(name);
+        if (field instanceof Blockly.FieldVariable) {
+          // TODO (marisaleung): When we change setValue and getValue to
+          // interact with id's instead of names, update this so that we get
+          // the variable based on id instead of textContent.
+          var variable = workspace.getVariable(xmlChild.textContent);
+          var type = xmlChild.getAttribute('variabletype');
+          if (type || type == '') {
+            if (type !== variable.type) {
+              throw Error('Serialized variable type with id \'' +
+                variable.getId() + '\' had type ' + variable.type + ', and ' +
+                'does not match variable field that ' + 'references it: ' +
```

Comment:
unnecessary string concat 'does not match variable field that ' + 'references it'